src/main.rs

```diff
@@ -314,5 +348,47 @@ fn draw(
             .chain(once(list_view_widget)),
     );
 
-    state.update_skip_offset(rx.recv().unwrap());
+    let info: ListViewInfo = rx.recv().unwrap();
+    state.update_skip_offset(info.new_skip);
+
+    if surface.is_auto_height() {
+        let mut full_height = info.new_y
+            + list_config.font_size as u32
+            + list_config.margin.bottom as u32
+            + ADDITIONAL_CAP;
+
+        if animator.contains("HeightAnimation") {
+            surface.commit();
+            return;
+        }
+
+        if full_height != old_height {
+            if full_height > background_config.height {
+                full_height = background_config.height;
+            }
+
+            // Hack for input_changed: If input buffer capacity is 0 then we do not typing
+            // anything, so disable animation for first calls of draw (for support
+            // ListItem::show_default)
```

Comment:
1. This hack is inside `input_changed`, right?
2. Why it's capacity and not length?
3. Actually animation is added here, why it stated it's disabled then?

Reply:
Yes, this hack inside `input_changed`. Length of string will be zero if we are typing something and delete it. But capacity will not be zero anymore because we don't free string. It is for first draw of `yofi`.
Render:
1. Run yofi
2. Draw window with `height` from `Config`
3. Try to place `list_items` in here
4. Truncate `height` to new height with removing unused height space. If it is just a first render we do not want to see this "jumping" window, but only `input`, thats why i added this